test/e2e/proxy_test.go

```diff
@@ -399,58 +577,70 @@ func createAgnhostPod(t *testing.T, data *TestData, podName string, node string,
 func testNodePortClusterFromRemote(t *testing.T, data *TestData, nodes, urls []string) {
 	skipIfKubeProxyEnabled(t, data)
 	for idx, node := range nodes {
-		require.NoError(t, probeFromNode(node, urls[idx]), "Service NodePort whose externalTrafficPolicy is Cluster should be able to be connected from remote Node")
+		require.NoErrorf(t, probeFromNode(node, urls[idx]),
+			"Service NodePort should be able to be connected from remote Node %s with URL %s",
+			node, urls[idx])
 	}
 }
 
 func testNodePortClusterFromNode(t *testing.T, data *TestData, nodes, urls []string) {
 	skipIfKubeProxyEnabled(t, data)
 	for idx, node := range nodes {
-		require.NoError(t, probeFromNode(node, urls[idx]), "Service NodePort whose externalTrafficPolicy is Cluster should be able to be connected from Node")
+		require.NoErrorf(t, probeFromNode(node, urls[idx]),
+			"Service NodePort should be able to be connected from Node %s with URL %s",
+			node, urls[idx])
 	}
 }
 
-func testNodePortClusterFromPod(t *testing.T, data *TestData, pods, urls []string) {
+func testNodePortClusterFromPod(t *testing.T, data *TestData, pods, podOSes, urls []string) {
 	for _, url := range urls {
-		for _, pod := range pods {
-			require.NoError(t, probeFromPod(data, pod, url), "Service NodePort whose externalTrafficPolicy is Cluster should be able to be connected from Pod")
+		for idx, pod := range pods {
+			require.NoErrorf(t, probeFromPod(data, pod, podOSes[idx], url),
+				"Service NodePort should be able to be connected from Pod '%s' with URL '%s'", pod, url)
 		}
 	}
 }
 
-func testNodePortLocalFromRemote(t *testing.T, data *TestData, nodes, urls, expectedClientIPs, expectedHostnames []string) {
+func testNodePortLocalFromRemote(t *testing.T, data *TestData, nodes, urls, expectedClientIPs []string, expectedHostNetwork string) {
 	skipIfKubeProxyEnabled(t, data)
-	errMsg := "Service NodePort whose externalTrafficPolicy is Local should be able to be connected from remote Node"
+	errMsg := "Service NodePort should be able to be connected from remote Node"
 	for idx, node := range nodes {
-		hostname, err := probeHostnameFromNode(node, urls[idx])
-		require.NoError(t, err, errMsg)
-		require.Equal(t, hostname, expectedHostnames[idx])
+		// It hangs on Linux Pod/Node to execute <Windows-nodeIP>/shell?cmd=<some-command> while clientip/hostname endpoints work.
+		if len(clusterInfo.windowsNodes) == 0 || nodeOS(node) != "linux" {
+			hostNetwork, err := probeFromNodeWithEndpoint(node, urls[idx], hostNetworkNodeEP)
+			require.NoError(t, err, errMsg)
+			require.Equal(t, expectedHostNetwork, hostNetwork,
+				"unexpected hostNetwork on Node '%s' with URL '%s/%s'", node, urls[idx], hostNetworkNodeEP)
```

Comment:
The `require.Equal` assertions in these tests should probably be `assert.Equal` assertions instead, as there is no need for the test to stop immediately if one of them fails.

Reply:
Makes sense. Updated.